src/main/java/training/chessington/model/pieces/Bishop.java

```diff
@@ -15,11 +12,6 @@ public Bishop(PlayerColour colour, Board board) {
 
     @Override
     public List<Move> getAllowedMoves(Coordinates from) {
-        return new ArrayList<>();
-    }
-
-    @Override
-    protected boolean moveIsValid(Move move) {
-        return true;
+        return new ArrayList<>(moveUtil.getDiagonalMoves(from, super.colour));

```

Comment:
Do you need the `super.`?

Reply:
No, I just thought it made it clearer that that's where the colour field is.

Reply:
Ah, I see. It's not idiomatic to do that in Java, particularly when IDEs are so good now (you can Ctrl-click on anything to see where it's come from).